stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -275,31 +294,34 @@ std::vector<cav_msgs::TrajectoryPlanPoint> StopandWait::compose_trajectory_from_
   std::vector<lanelet::BasicPoint2d> raw_points;
   splitPointSpeedPairs(final_points, &raw_points, &speeds);
 
-  std::vector<double> downtracks = carma_wm::geometry::compute_arc_lengths(raw_points);
+  // Convert the inverted downtracks back to regular downtracks
+  double max_downtrack = inverse_downtracks.back();
+  std::vector<double> downtracks = lanelet::utils::transform(inverse_downtracks, [max_downtrack](const auto& d) { return max_downtrack - d; });
+  std::reverse(downtracks.begin(),
+               downtracks.end()); 
```

Comment:
Are we sure that we are not "reverting" them first by changing downtrack values, and the reverting them again by std::reverse functon?

Reply:
Yes. If you work it out by hand, the inverse one has both reversed indexes and inverted counts, so we need to fix the counts then correct the indexes in two steps.

Reply:
ah makes sense